cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewServer.java

```diff
@@ -0,0 +1,287 @@
+      new NamespaceStoreModule().getStandaloneModules(),
+      new RemoteSystemOperationsServiceModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          bind(ArtifactRepository.class).toInstance(artifactRepository);
+          bind(ArtifactStore.class).toInstance(artifactStore);
+          bind(AuthorizerInstantiator.class).toInstance(authorizerInstantiator);
+          bind(AuthorizationEnforcer.class).toInstance(authorizationEnforcer);
+          bind(PrivilegesManager.class).toInstance(privilegesManager);
+          bind(StreamAdmin.class).toInstance(streamAdmin);
+          bind(StreamConsumerFactory.class).toInstance(streamConsumerFactory);
+          bind(StreamCoordinatorClient.class).toInstance(streamCoordinatorClient);
+          // bind explore client to mock.
+          bind(ExploreClient.class).to(MockExploreClient.class);
+        }
+      });
+  }
+
+  /**
+   * starts preview services
+   */
+  @Override
+  protected void startUp() throws Exception {
+    metricsCollectionService.startAndWait();
+    datasetService.startAndWait();
+
+    // It is recommended to initialize log appender after datasetService is started,
+    // since log appender instantiates a dataset.
+    logAppenderInitializer.initialize();
+
+    LoggingContextAccessor.setLoggingContext(new ServiceLoggingContext(Id.Namespace.SYSTEM.getId(),
+                                                                       Constants.Logging.COMPONENT_NAME,
+                                                                       Constants.Service.PREVIEW_HTTP));
+    Futures.allAsList(
+      applicationLifecycleService.start(),
+      systemArtifactLoader.start(),
+      programRuntimeService.start(),
+      programLifecycleService.start()
+    ).get();
+
+    // Run http service on random port
+    httpService.startAndWait();
+
+    // cancel service discovery on shutdown
+    cancellable = discoveryService.register(ResolvingDiscoverable.of(new Discoverable() {
+      @Override
+      public String getName() {
+        return Constants.Service.PREVIEW_HTTP;
+      }
+
+      @Override
+      public InetSocketAddress getSocketAddress() {
+        return httpService.getBindAddress();
+      }
+    }));
+  }
+
+  @Override
+  protected void shutDown() throws Exception {
+    try {
+      cancellable.cancel();
+      httpService.stopAndWait();
+      programRuntimeService.stopAndWait();
+      applicationLifecycleService.stopAndWait();
+      systemArtifactLoader.stopAndWait();
+      programLifecycleService.stopAndWait();
+      logAppenderInitializer.close();
+      datasetService.stopAndWait();
+      metricsCollectionService.stopAndWait();
+    } finally {
+      cleanupTempDir();
+    }
+  }
+
+  private void cleanupTempDir() {
+    File tmpDir = new File(cConf.get(Constants.CFG_LOCAL_PREVIEW_DIR),
+                           cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
```

Comment:
If you want to have everything in the preview dir auto cleanup when CDAP starts, you can simple use a temp dir under `data/temp` (Use `Files.createTempDirectory` to create one under `data/temp`).